lib/screens/v2/candidate_generator/on_bus.ex

```diff
@@ -26,9 +27,14 @@ defmodule Screens.V2.CandidateGenerator.OnBus do
   end
 
   @impl CandidateGenerator
-  def candidate_instances(_config, query_params) do
+  def candidate_instances(
+        config,
+        query_params,
+        departures_instances_fn \\ &OnBus.Departures.departures_instances/3
+      ) do
     [
-      fn -> body_instances(query_params) end
+      fn -> body_instances(query_params) end,
+      fn -> departures_instances_fn.(config, query_params.route_id, query_params.stop_id) end
```

Comment:
Noting that there needs to be a check on the `route_id` and `stop_id` in case they do not exist on `query_params`, but I'll get to that later

Reply:
I am updating this to pass in the full `query_params` struct and then to pattern match on if `route_id` and `stop_id`